### ISSUE TITLE
TypeScript declaration file issue

### ISSUE THREAD

**Issue by alexmic** [jbielick/faktory_worker_node#51]
node.js version: 12.14.1
faktory-server version:  1.4.0
`faktory-worker` package version: latest

#### Code sample:

```js
worker.work(opts).catch((err) => {
  console.error("Worked crashed, exiting", err);
  process.exit(1);
});
```

#### Expected Behavior:
When compiling with `tsc` I expected the above to be compiled ok.

#### Actual Behavior:
It throws an error:
`error TS2339: Property 'catch' does not exist on type 'Worker'`
I think this is because the declaration file defines `Worker` as the return type instead of `Promise<Worker>`.

#### Steps to reproduce the behavior or a test case:
The faktory-work executable should be a good test case.

**Comment by jbielick** [comment]
Hey Alex,

I thought it would be interesting to see what this project looks like in typescript, so I added typing to the project. I think I'll continue to use typescript for this library. For now, the package is published as `faktory-worker@next` (`4.0.0-beta.1`). I think it's possible there could be some issues introduced (as well as fixed!), so you could try out the typescript version of this library with `npm install faktory-worker@next` if you want to try it.

Let me know if you have issues.

**Comment by jbielick** [comment]
I've also published `3.3.7` with fixes to the type declarations.

**Comment by alexmic** [comment]
Thanks for the quick turn-around! I will try it and report back.
